Integrations/integration-CheckpointFirewall

```diff
@@ -0,0 +1,707 @@
+            'task-name':                    'TaskName'
+        }
+    };
+
+    function extend(obj, src) {
+        Object.keys(src).forEach(function(key) { obj[key] = src[key]; });
+        return obj;
+    }
+
+    //returns single object withing entity (i.e. File[0])
+    var jsonToEntityObject = function(origObj, newKeys){
+        //log("func started");
+        var ret = {};
+        var path;
+        var newField;
+        //var temp;
+        for(var key in newKeys){
+            if(newKeys[key]){
+                path = newKeys[key].split('.');
+                origPath = key.split('.');
+                if(path.length == 1){
+                    ret[newKeys[key]] = dq(origObj, '.'+key);
+                }
+                else{
+                    newField = dq(origObj, '.'+key);
+                    if(!ret[path[0]] && newField){
+                        ret[path[0]] = {};
+                    }
+                    if(newField){
+                        ret[path[0]][path[1]] = newField;
+                    }
+                }
+            }
+        }
+        return ret;
+    };
+
+    //returns entire entity array (i.e. File)
+    var jsonToEntity = function(origObj, newKeys){
+        var j;
+        var ret;
+        if(!Array.isArray(origObj)){
+            ret = [jsonToEntityObject(origObj, newKeys)];
+            return ret;
+        }
+        else if(origObj.length > 0){ //makes sure no empty arrays are pushed
+            ret = [];
+            for(j=0; j<origObj.length; j++){
+                ret.push(jsonToEntityObject(origObj[j], newKeys));
+            }
+            return ret;
+        }
+    };
+
+    var sendRequest = function(cmdURL, body, cmdHeaders) {
+        var headers = cmdHeaders? cmdHeaders : {'content-type': ['application/json'], 'X-chkp-sid' : [sid]};
+        var res = http(
+            url+cmdURL,
+            {
+                Method: 'POST',
+                Headers: headers,
+                Body: body
+            },
+            params.insecure,
+            params.proxy
+        );
+        if (res.StatusCode < 200 || res.StatusCode >= 300) {
+            throw 'Request Failed.\nStatus code: ' + res.StatusCode + '.\nBody: ' + JSON.stringify(res) + '.';
+        }
+        return res;
+    };
+
+    var login = function(){
+        var res = sendRequest('login', JSON.stringify({'user':params.credentials.identifier,'password':params.credentials.password}), {'content-type': ['application/json']});
+        sid = JSON.parse(res.Body).sid;
+        return sid;
+    };
+
+    var logout = function(){
+        if(sid){
+            var res = sendRequest('logout', '{}');
+            sid = undefined;
+            return JSON.parse(res.Body).message;
+        }
+        return 'Already logged out';
+    };
+
+    var add_host = function(){
+        sendRequest('add-host', JSON.stringify({name:args.ip, 'ip-address':args.ip}));
+        sendRequest('publish', '{}');
```

Comment:
every change you do in checkpoint (add, set etc.) needs a publish command to save the outcome. Without it objects get locked by the session and access to them is locked.